set.mm

```diff
@@ -147314,83 +147758,180 @@ of an open range of nonnegative integers (of length equal to the length of
+    HWSVFVGWDVHZWDVIGWDVJVKWSXGWDVLTWDGWDVMVNWSXIXDXJXIXDWDVOTMVPMABEWEWDVQNWAW
+    LBQWBABVRPVS $.
+
+  $( The last symbol of a word concatenated with the word with the last symbol
+     removed having results in the word itself.  (Contributed by Alexander van
+     der Vekens, 24-Oct-2018.)  (Revised by AV, 9-May-2020.) $)
+  ccats1pfxeq $p |- ( ( W e. Word V /\ U e. Word V
+                         /\ ( # ` U ) = ( ( # ` W ) + 1 ) )
+                       -> ( W = ( U prefix ( # ` W ) )
+                            -> U = ( W ++ <" ( lastS ` U ) "> ) ) ) $=
+    ( wcel chash cfv c1 co wceq cpfx cconcat oveq1 syl eqcomd 3ad2ant1 3ad2ant3
+    cmin eqtrd cc0 clt cword caddc w3a cs1 wa adantl lencl nn0cnd pncan1 oveq2d
+    clsw cc oveq1d c0 wne simp2 wbr cn0 nn0p1gt0 breq2 mpbird hashneq0 3ad2ant2
+    wb mpbid pfxlswccat syl2anc adantr eqtr2d ex ) CBUAZDZAVKDZAEFZCEFZGUBHZIZU
+    CZCAVOJHZIZACAUKFUDZKHZIVRVTUEWBVSWAKHZAVTWBWCIVRCVSWAKLUFVRWCAIVTVRWCAVNGQ
+    HZJHZWAKHZAVRVSWEWAKVRVOWDAJVRVOVPGQHZWDVLVMVOWGIVQVLWGVOVLVOULDWGVOIVLVOBC
+    UGZUHVOUIMNOVQVLWGWDIVMVQWDWGVNVPGQLNPRUJUMVRVMAUNUOZWFAIVLVMVQUPVRSVNTUQZW
+    IVRWJSVPTUQZVLVMWKVQVLVOURDWKWHVOUSMOVQVLWJWKVDVMVNVPSTUTPVAVMVLWJWIVDVQAVK
+    VBVCVEBAVFVGRVHVIVJ $.
+
+  $( Obsolete proof of ~ ccats1pfxeq as of 12-Oct-2022.  (Contributed by
+     Alexander van der Vekens, 24-Oct-2018.)
+     (Proof modification is discouraged.)  (New usage is discouraged.) $)
+  ccats1swrdeqOLD $p |- ( ( W e. Word V /\ U e. Word V
                          /\ ( # ` U ) = ( ( # ` W ) + 1 ) )
                          -> ( W = ( U substr <. 0 , ( # ` W ) >. )
                               -> U = ( W ++ <" ( lastS ` U ) "> ) ) ) $=
     ( wcel chash cfv c1 co wceq cc0 cop csubstr cconcat oveq1 3ad2ant3 3ad2ant1
     cmin eqtr2d clt wbr cword caddc w3a cs1 wa adantl cn0 cc lencl nn0cn pncan1
     clsw 3syl opeq2d oveq2d oveq1d c0 wne simp2 nn0p1gt0 syl wb mpbird hashneq0
-    breq2 3ad2ant2 mpbid swrdccatwrd syl2anc eqtrd adantr ex ) CBUAZDZAVMDZAEFZ
-    CEFZGUBHZIZUCZCAJVQKZLHZIZACAULFUDZMHZIVTWCUEWEWBWDMHZAWCWEWFIVTCWBWDMNUFVT
-    WFAIWCVTWFAJVPGQHZKZLHZWDMHZAVTWBWIWDMVTWAWHALVTVQWGJVTWGVRGQHZVQVSVNWGWKIV
-    OVPVRGQNOVNVOWKVQIZVSVNVQUGDZVQUHDWLBCUIZVQUJVQUKUMPRUNUOUPVTVOAUQURZWJAIVN
-    VOVSUSVTJVPSTZWOVTWPJVRSTZVNVOWQVSVNWMWQWNVQUTVAPVSVNWPWQVBVOVPVRJSVEOVCVOV
-    NWPWOVBVSAVMVDVFVGBAVHVIVJVKRVL $.
+    breq2 3ad2ant2 mpbid swrdccatwrdOLD syl2anc eqtrd adantr ex ) CBUAZDZAVMDZA
+    EFZCEFZGUBHZIZUCZCAJVQKZLHZIZACAULFUDZMHZIVTWCUEWEWBWDMHZAWCWEWFIVTCWBWDMNU
+    FVTWFAIWCVTWFAJVPGQHZKZLHZWDMHZAVTWBWIWDMVTWAWHALVTVQWGJVTWGVRGQHZVQVSVNWGW
+    KIVOVPVRGQNOVNVOWKVQIZVSVNVQUGDZVQUHDWLBCUIZVQUJVQUKUMPRUNUOUPVTVOAUQURZWJA
+    IVNVOVSUSVTJVPSTZWOVTWPJVRSTZVNVOWQVSVNWMWQWNVQUTVAPVSVNWPWQVBVOVPVRJSVEOVC
+    VOVNWPWOVBVSAVMVDVFVGBAVHVIVJVKRVL $.
+
+  ${
+    $d U s $.  $d V s $.  $d W s $.
+    $( There exists a symbol such that its concatenation with the prefix
```

Comment:
"its concatenation AFTER" is clearer: on first reading, I just thougth "how can this be true?" since I assumed "EARLIER" was meant.
The statement looks overly complex, but maybe it's used as a lemma in this form later.

Reply:
Done